pkg/detector/detector.go

```diff
@@ -253,8 +254,10 @@ func (d *ResourceDetector) EventFilter(obj interface{}) bool {
 	}
 
 	// if SkippedPropagatingNamespaces is set, skip object events in these namespaces.
-	if _, ok := d.SkippedPropagatingNamespaces[clusterWideKey.Namespace]; ok {
-		return false
+	for _, ns := range d.SkippedPropagatingNamespaces {
+		if match, _ := regexp.MatchString(fmt.Sprintf("^%s$", ns), clusterWideKey.Namespace); match {
+			return false
+		}
 	}
```

Comment:
Given the `EventFileter()` will be called by each event, I'm concerned about the performance, as far as I know, the regular expression might be low.

I'm thinking should we add a benchmark test?